lib/cstruct.ml

```diff
@@ -282,13 +283,923 @@ let get_uint64 swap p t i =
     let r = ba_get_int64 t.buffer (t.off+i) in
     if swap then swap64 r else r [@@inline]
 
+(* o - bitoffset, i - count of bits *)
+(* It assumes we use __attribute__((packed)) with alignment = 1 byte *)
```

Comment:
Does this assumption hold in general? What are the alternatives? Should this be configurable? :slightly_smiling_face:

Reply:
I could make it configurable, then it would become not +50%, but +500% of additional code.

Reply:
The only viable alternative if we do not assume this is to switch to using ocaml-ctypes (which does handle real C alignments) in order to determine a struct layout (see [my earlier comment](https://github.com/mirage/ocaml-cstruct/pull/215#issuecomment-451433761)).

I'm in two minds about how to handle this usecase -- an alternative approach to solve the problem of bitfields is to define a ctypes struct layout, do the stub generation/probing that ctypes does already, and then re-expose those offsets using the Cstruct conventions.  That would involve actually removing almost all the code in ppx_cstruct since the C compiler would do the bulk of the work then :-)